### DISCUSSION TITLE
Accessibility in DHIS2 apps and libraries

### DISCUSSION THREAD

**Discussion by paschalidi** [dhis2/notes#211] [Ideas]
_"Accessible design not only helps users with special abilities; it provides better user experiences for our users in general."_ 

I am writing this issue driven by the assumption that we see the benefits of creating an accessible  design. Therefore this issue will serve as an open discussion regarding accessibility and where our stand on the topic is. We can evaluate the effort, discuss and organise how we would like to tackle the topic.

Some useful resources: 
* [Web Content Accessibility Guidelines (WCAG) 2.1](https://www.w3.org/TR/WCAG21/), this is a huge document that covers a wide range of recommendations for making Web content more accessible. 
* [Carbon Design System](https://www.carbondesignsystem.com/guidelines/accessibility/overview/), these people do a very good job in keeping their components accessible for the web. They also have tons of good documentation and suggestions on how to do so. 
* There are plenty of tooling on how you could have automated checks on your CI/CD pipeline. Some of those would be [web-hint](https://webhint.io/), [wave](https://wave.webaim.org/), [aXe](https://github.com/dequelabs/axe-core), [Pa11y](https://github.com/pa11y/pa11y), [Koa11y](https://open-indy.github.io/Koa11y/), [Google Accessibility Developer Tools](https://github.com/GoogleChrome/accessibility-developer-tools), [AATT aka paypal's solution](https://github.com/paypal/AATT) 

I am sure there many more examples and resources out there I am simply citing the ones I know of.

**Comment by paschalidi** [comment]
I just found this [article](https://stegosource.com/how-to-build-html-forms-right-accessibility) that might be relevant regarding forms and accessibility. I will just leave it here we might wanna take some ideas later on.

**Comment by ghost** [comment]
See also this slack thread: https://dhis2.slack.com/archives/C0BP0RABF/p1589444680027100 about an accessibility linter.

**Comment by amcgee** [comment]
Another accessibility automation library https://github.com/IBMa/equal-access

**Comment by paschalidi** [comment]
@amcgee I was thinking of this the other day. Perhaps we can also include this on our weekly meetings? What you think?
